hdfs/client.py

```diff
@@ -308,6 +309,22 @@ def set_acl(self, hdfs_path, acl_spec):
     _logger.info('Setting ACL spec for %r to %r.', hdfs_path, acl_spec)
     self._set_acl(hdfs_path, aclspec=acl_spec)
 
+  def modify_acl(self, hdfs_path, acl_spec):
```

Comment:
Since this function is very similar to the existing `set_acl`, let's combine the two. For example we could add a `clear` boolean flag (defaulting to `True`) to toggle between the two behaviors.